### PR TITLE
My display events

### PR DESCRIPTION
**Pull request by michaelnyu**
Added a components folder which holds both components and styles for each component.  

Added an EVENTITEM component and a EVENTLIST component.  

Set up modular style sheets and a global style sheet ( commonstyles.js ).  

Cleaned up some of the spacing issues in App.js.  

Implemented a displayEvents page in App.js.  (unsure of whether to branch each view into a separate container)
  - HomeContainer class now contains a conditional navigate to choose between 2 different views based off the existence of events.
  - Added a temporary "events array" to mimic the result of hitting our API endpoint

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
 
 	];
+
+
+class HomeScreen extends React.Component {
+	constructor() {
+		super();
+		this.state = {
+   			eventsArray: ["events1", "events2"],
+      	}
```
Comment (epicchewy) [inline App.js]: nit: tab

File: App.js [hunk 2]
```diff
@@ -27,41 +30,73 @@ let allNames = [
 		}
 
 	];
+
+
+class HomeScreen extends React.Component {
+	constructor() {
```
Comment (epicchewy) [inline App.js]: format: 2-space tabs

File: App.js [hunk 3]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
+      	}
+	}
 	
+	componentWillMount() {
+		// fetch call goes here to fill eventsArray
+  	}
 
+ 	render() {
+		const { navigate } = this.props.navigation; 
+		let eventsArray = this.state.eventsArray;
```
Comment (epicchewy) [inline App.js]: use const for immutable

File: App.js [hunk 4]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
+		let pic = {
+		uri: 'https://pbs.twimg.com/profile_images/862164234947440640/WqQ358Yw_400x400.jpg'
```
Comment (epicchewy) [inline App.js]: tab

File: App.js [hunk 5]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
+		return (
+			<View style={styles.container} >
+				<Image source={pic} style={styles.picture} />
+				<Text style={styles.titleText}>Hello BE!</Text>
+
+				<Button onPress={() => {
+			  		if (eventsArray.length > 0){
+			  			navigate('DisplayEvents', { eventsArray: eventsArray });
+			  		}
+			  		else{
```
Comment (epicchewy) [inline App.js]: put else on same line

File: App.js [hunk 6]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
+			  			navigate('Profile');
+			  		}
+			  	}} 
+				title="Next" />
+		  	</View>
+		);
+  	}
+}
 
+class DisplayEventsScreen extends React.Component {
```
Comment (epicchewy) [inline App.js]: one component per file? i feel like its easier to maintain that way

File: App.js [hunk 7]
```diff
@@ -27,41 +30,73 @@ let allNames = [
...
+	constructor(props) {
+		super(props);
+	}
 
-class HomeScreen extends React.Component {
-	contructor(props) {
+	render() {
+		const { navigate } = this.props.navigation;
+		let {eventsArray} = this.props.navigation.state.params;
```
Comment (epicchewy) [inline App.js]: const { eventsArray }

File: App.js [hunk 8]
```diff
@@ -87,105 +122,101 @@ class CreationScreen extends React.Component {
...
+	render() {
+		const { navigate } = this.props.navigation;
+		const curMoment = moment(this.props.navigation.state.params.time).format("YYYY/DD/MM")
+		return (
+		  <View style={styles.container} >
+			 <Text style={styles.creationSubText}> Time</Text>
+			 <Text style={styles.creationSubText}> {curMoment} </Text>
+			 <Text style={styles.creationSubText}> Activity Type </Text>
+			 <Text style={styles.creationSubText}> {this.props.navigation.state.params.activity} </Text>
+			 <TextInput style={{height: 40, width: 200, borderColor: 'black', borderWidth: 2, }}  value={this.state.inputText} onChangeText={(text) => this.setState({ inputText: text })} />
```
Comment (epicchewy) [inline App.js]: no parentheses for single argument arrow functions `onChangeTest={text => ... }`

File: App.js [hunk 9]
```diff
@@ -87,105 +122,101 @@ class CreationScreen extends React.Component {
...
+	Home: { screen: HomeScreen },
 	Profile: { screen: CreationScreen },
 	Invite: {screen: InviteScreen }, 
-    });
-
-
+	DisplayEvents: {screen: DisplayEventsScreen},
+});
 
 export default App;
```
Comment (epicchewy) [inline App.js]: nit: new line

File: commonstyles.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export const NAVY = '#072956';
+export const BLUE = '#527edd';
+export const CREAM = '#f5f1ed';
+export const FONT_NORMAL = 'Montserrat-Regular';
+export const FONT_BOLD = 'OpenSans-Bold';
+export const BORDER_RADIUS = 5;
```
Comment (epicchewy) [inline commonstyles.js]: nit: new line

File: components/eventitem.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import styles from './eventitemstyle';
+import {
+    Text, View, TouchableHighlight
```
Comment (epicchewy) [inline components/eventitem.js]: just a note: try to be consistent with when you use multiline imports. use your best judgement for when the import is too long/unreadable

File: components/eventitemstyle.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+		fontSize: 30,
+		fontWeight: 'bold',
+		textAlign: 'center',
+		marginTop: 20,
+		marginBottom: 20,
+	},
+	navy: {
+		color: NAVY,
+	}
+});
```
Comment (epicchewy) [inline components/eventitemstyle.js]: nit: new line

File: components/eventitem.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import styles from './eventitemstyle';
```
Comment (epicchewy) [inline components/eventitem.js]: move all styling to its own folder /components/style, /style or something like that

File: components/eventliststyles.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
+import React, { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
+	list: {
+		height: 500,
+	},
+});
```
Comment (epicchewy) [inline components/eventliststyles.js]: nit: new line

File: components/eventitemstyle.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import React, { StyleSheet } from 'react-native';
+import { NAVY } from '../commonstyles';
+export default StyleSheet.create({
```
Comment (epicchewy) [inline components/eventitemstyle.js]: add line above export

**Review by epicchewy — CHANGES_REQUESTED** [review]

File: App.js [hunk 16]
```diff
@@ -27,41 +30,73 @@ let allNames = [
 		}
 
 	];
+
+
+class HomeScreen extends React.Component {
+	constructor() {
+		super();
+		this.state = {
+   			eventsArray: ["events1", "events2"],
```
Comment (epicchewy) [inline App.js]: move dummy data somewhere else

File: App.js [hunk 17]
```diff
@@ -2,66 +2,106 @@ import React from 'react';
...
+		ImageLink: {pic2}, 
+	},
+	{
+		name: 'Rahul Sheth',
+		ImageLink: {pic2}, 
+	}, 
+	{
+		name: 'Hamilton Tran', 
+		ImageLink: {pic2}, 
+	},
```
Comment (epicchewy) [inline App.js]: can we export this dummy data to another file?

File: App.js [hunk 18]
```diff
@@ -2,66 +2,106 @@ import React from 'react';
...
+		// fetch call goes here to fill eventsArray
+	}
+
+	render() {
+		const { navigate } = this.props.navigation; 
+		const eventsArray = this.state.eventsArray;
+		const pic = {
+			uri: 'https://pbs.twimg.com/profile_images/862164234947440640/WqQ358Yw_400x400.jpg'
+		}; 
+		return (
```
Comment (epicchewy) [inline App.js]: new line here for readability

File: App.js [hunk 19]
```diff
@@ -2,66 +2,106 @@ import React from 'react';
...
+			<View style={styles.container} >
+				<Image source={pic} style={styles.picture} />
+				<Text style={styles.titleText}>Hello BE!</Text>
+
+				<Button onPress={() => {
```
Comment (epicchewy) [inline App.js:43]: not sure if the () are optional or not for 0 argument lambda

File: App.js [hunk 20]
```diff
@@ -2,66 +2,106 @@ import React from 'react';
...
+
+/*****************************************/
+/* Global Variables for Testing Purposes */
+/*****************************************/
+
+const moment = require('moment');
+const idLocale = require('moment/locale/id'); //for Indonesia locale
 moment.locale('id', idLocale);
+
 let pic2 = {
```
Comment (epicchewy) [inline App.js]: two space indents? that is the js standard

File: App.js [hunk 21]
```diff
@@ -87,105 +127,102 @@ class CreationScreen extends React.Component {
...
-	         <List containerStyle={{marginBottom: 20, width: 200}}>
-	         	{
-	         		allNames.map((l, i) => (
-	         			<ListItem 
-	         				key={i}
-	         				title={l.name}
+	render() {
+		const { navigate } = this.props.navigation;
+		const curMoment = moment(this.props.navigation.state.params.time).format("YYYY/DD/MM")
+		return (
```
Comment (epicchewy) [inline App.js]: nit: new line

File: App.js [hunk 22]
```diff
@@ -87,105 +127,102 @@ class CreationScreen extends React.Component {
...
-    );
+		  </View>
+	);
   }
 }
 
 
 
+
 const styles = StyleSheet.create({
```
Comment (epicchewy) [inline App.js:155]: is it standard practice to keep styles and view together for react native? if not, lets keep them decoupled

File: common_styles.js [hunk 23]
```diff
@@ -0,0 +1,6 @@
+export const NAVY = '#072956';
+export const BLUE = '#527edd';
+export const CREAM = '#f5f1ed';
+export const FONT_NORMAL = 'Montserrat-Regular';
+export const FONT_BOLD = 'OpenSans-Bold';
+export const BORDER_RADIUS = 5;
```
Comment (epicchewy) [inline common_styles.js:6]: nit: new line

File: App.js [hunk 24]
```diff
@@ -87,105 +127,102 @@ class CreationScreen extends React.Component {
...
+			 <Text style={styles.creationSubText}> {this.props.navigation.state.params.activity} </Text>
+			 <TextInput style={{height: 40, width: 200, borderColor: 'black', borderWidth: 2, }}  value={this.state.inputText} onChangeText={text => this.setState({ inputText: text })} />
+
+			 <List containerStyle={{marginBottom: 20, width: 200}}>
+				{
+					allNames.map((l, i) => (
+						<ListItem 
+							key={i}
+							title={l.name}
 					onPress={(person) => this.setState({ namesArray: [...this.state.namesArray, l.name, ', '] })}
```
Comment (epicchewy) [inline App.js:140]: no parentheses around single argument double arrow functions

File: components/event_item.js [hunk 25]
```diff
@@ -0,0 +1,25 @@
...
+    Text, 
+    View, 
+    TouchableHighlight,
+} from 'react-native';
+
+class EventItem extends React.Component {
+    constructor(){
+        super();
+    }
+    render() {
```
Comment (epicchewy) [inline components/event_item.js:14]: nit: new line

File: components/event_item.js [hunk 26]
```diff
@@ -0,0 +1,25 @@
...
+
+        return (
+            <View style={styles.eventitem}>
+                <Text style={[styles.name, styles.navy]}>{name}</Text>
+            </View>
+        );
+    }
+}
+
+module.exports = EventItem;
```
Comment (epicchewy) [inline components/event_item.js:25]: nit: new line

File: components/event_list.js [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+    constructor(){
+        super();
+    }
+
+    render() {
+        const eventsArray = this.props.events;
+
+        return (
+            <View style={styles.list}>
+            {eventsArray.map((event, i) =>
```
Comment (epicchewy) [inline components/event_list.js:20]: nit: indent this block

File: components/styles/event_item_style.js [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+		fontWeight: 'bold',
+		textAlign: 'center',
+		marginTop: 20,
+		marginBottom: 20,
+	},
+	navy: {
+		color: NAVY,
+	},
+}
+);
```
Comment (epicchewy) [inline components/styles/event_item_style.js]: nit: new line

File: components/styles/event_list_styles.js [hunk 29]
```diff
@@ -0,0 +1,8 @@
+import React, { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
+	list: {
+		height: 500,
+	},
+}
+);
```
Comment (epicchewy) [inline components/styles/event_list_styles.js]: nit: new line

File: screens/DisplayEventsScreen.js [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+import EventList from '../components/eventlist';
+
+class DisplayEventsScreen extends React.Component {
+	constructor(props) {
+		super(props);
+	}
+
+	render() {
+		const { navigate } = this.props.navigation;
+		let { eventsArray } = this.props.navigation.state.params;
```
Comment (epicchewy) [inline screens/DisplayEventsScreen.js]: const here?

File: screens/NavigationBar.js [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+export default class NavigationExampleScreen extends Component {
+  state = {};
+
+  render() {
+    return (
+      <Screen>
+        <Navigator.Config {...this.state} />
+        <Row
+          title="Title"
+          onPress={() => this.setState({
```
Comment (epicchewy) [inline screens/NavigationBar.js]: same as above

File: screens/NavigationBar.js [hunk 32]
```diff
@@ -0,0 +1,72 @@
...
+            title: 'A title',
+            subtitle: undefined,
+            rightButtons: undefined,
+          })}
+        />
+        <Row
+          title="Title and subtitle"
+          onPress={() => this.setState({
```
Comment (epicchewy) [inline screens/NavigationBar.js]: same as above

File: screens/NavigationBar.js [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+          })}
+        />
+      </Screen>
+    );
+  }
+}
+
+NavigationExampleScreen.defaultProps = defaultProps;
+NavigationExampleScreen.propTypes = propTypes;
+NavigationExampleScreen.contextTypes = contextTypes;
```
Comment (epicchewy) [inline screens/NavigationBar.js]: nit: new line

**Review by epicchewy — CHANGES_REQUESTED** [review]
much better, but lets try to remain consistent with style throughout the entire repo

File: App.js (hunk 19 above)
Comment (michaelnyu) [reply App.js:43]: it is

File: App.js (hunk 20 above)
Comment (michaelnyu) [reply App.js]: change all files to be a two space indent?

Comment (epicchewy) [reply App.js]: yes

File: common_styles.js (hunk 23 above)
Comment (michaelnyu) [reply common_styles.js:6]: what does nit mean

File: App.js (hunk 22 above)
Comment (michaelnyu) [reply App.js:155]: i agree

File: App.js (hunk 19 above)
Comment (epicchewy) [reply App.js:43]: if they're optional, lets remove them to be `{ => { ... }}`

File: common_styles.js (hunk 23 above)
Comment (epicchewy) [reply common_styles.js:6]: nit pick

File: App.js (hunk 19 above)
Comment (michaelnyu) [reply App.js:43]: i meant it is necessary.  my b

**Pull request closed without merging**